plutus-benchmark/bls-benchmarks/exe/Main.hs

```diff
@@ -0,0 +1,161 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE NumericUnderscores #-}
+{-# OPTIONS_GHC -Wno-unused-top-binds #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+{- | Print out the costs of various test scripts involving the BLS12_381
+   primitives.  Most of these work on varying numbers of inputs so that we can
+   get an idea of what we can do within the on-chain execution limits.
+-}
+module Main (main)
+
+where
+
+import PlutusBenchmark.BLS12_381.Common
+
+import PlutusCore.Evaluation.Machine.ExBudget (ExBudget (exBudgetCPU, exBudgetMemory))
+import PlutusCore.Evaluation.Machine.ExBudgetingDefaults qualified as PLC
+import PlutusCore.Evaluation.Machine.ExMemory (ExCPU (..), ExMemory (..))
+import PlutusTx.Prelude as Tx hiding (sort, (*))
+import UntypedPlutusCore qualified as UPLC
+import UntypedPlutusCore.Evaluation.Machine.Cek qualified as Cek
+
+import Data.ByteString qualified as BS
+import Flat qualified
+import Text.Printf (printf)
+
+import Prelude (Double, IO, Integral, String, fromIntegral, mapM_, show, (*), (/))
+
+-- Protocol parameters (November 2022)
```

Comment:
This is duplicated from somewhere else, right?

Reply:
Yes, from `plutus-benchmark/ed25519-throughput/Main.hs`.  There's a ticket to put these somewhere global, but it hasn't been done yet.